### PR TITLE
Make "Deploy a project" docs vendor-neutral for Cloud IDE

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/1975
Resolves https://github.com/astronomer/docs/issues/2002

Release notes to follow

### CODE REVIEW

File: astro/cloud-ide/deploy-project.md [hunk 1]
```diff
@@ -34,101 +34,107 @@ Astro Cloud IDE projects use the same structure as Astro projects created with t
...
 
-On the first GitHub commit of your project, the Astro Cloud IDE automatically pushes a Github action named `astro_deploy.yaml` to your repository that includes steps for deploying to Astro. You can then configure your GitHub repository to push your Astro Cloud IDE project to a Deployment when you commit to specific branches. 
+## Deploy a project from a Git repository to Astro
 
-You can commit changes from the Astro Cloud IDE to your GitHub repository without configuring destination Deployments in GitHub. However, if you don't configure any destination Deployment, `astro_deploy.yaml` fails and your changes aren't deployed to Astro.
+Follow these steps to deploy to Astro whenever you merge commits to a `main` or a `dev` branch on your Git repository. You can modify these steps and the workflow to deploy to any number of Deployments from any number of branches.
```
Comment (jwitz) [inline astro/cloud-ide/deploy-project.md:85]:
```suggestion
Follow these steps to deploy to Astro whenever you merge commits to a `main` or a `dev` branch on the Git repository hosting your Astro Cloud IDE project. You can modify these steps and the workflow to deploy to any number of Deployments from any number of branches.
```

File: astro/cloud-ide/deploy-project.md [hunk 2]
```diff
@@ -34,101 +34,107 @@ Astro Cloud IDE projects use the same structure as Astro projects created with t
...
+After you configure a Git repository, you can use the **Commit** button in the Cloud IDE to commit your changes to your repository. 
 
 1. Click **Commit**.
 2. Select a branch type in the **Branch Type** list and then enter a name for the branch in the **Branch Name** field.
-3. In the **Commit Message** field, enter a commit message. This is the commit message for changes committed  from the Astro Cloud IDE to GitHub.
+3. In the **Commit Message** field, enter a commit message. This is the commit message for changes committed from the Astro Cloud IDE to your Git vendor.
 4. Click **Commit**.
 
-## Deploy a project from GitHub to Astro
+When you first make a commit to your repository, the Cloud IDE commits an additional workflow file that includes steps for pushing your code to Astro. You can then configure your GitHub repository to push your code to a Deployment whenever you commit to specific branches from the Astro Cloud IDE.
```
Comment (jwitz) [inline astro/cloud-ide/deploy-project.md:79]: What is the name of the action file on GitLab?

File: astro/cloud-ide/deploy-project.md [hunk 3]
```diff
@@ -34,101 +34,107 @@ Astro Cloud IDE projects use the same structure as Astro projects created with t
...
+    - `PROD_ASTRONOMER_KEY_SECRET` = `<your-prod-api-key-secret>`
+    - `PROD_ASTRONOMER_DEPLOYMENT_ID` = `<your-prod-astro-deployment-id>`
+    - `DEV_ASTRONOMER_KEY_ID` = `<your-dev-api-key-id>`
+    - `DEV_ASTRONOMER_KEY_SECRET` = `<your-dev-api-key-secret>`
+    - `DEV_ASTRONOMER_DEPLOYMENT_ID` = `<your-dev-astro-deployment-id>`
 
-After configuring your Github repository, commits from the Cloud IDE to your `main` or `dev` branches are automatically deployed to Astro.
+    See [GitHub](https://docs.github.com/en/actions/learn-github-actions/variables) and [GitLab](https://docs.gitlab.com/ee/ci/variables/) documentation on setting environment variables.  
+
+After configuring your repository, commits from the Cloud IDE to your `main` or `dev` branches are automatically deployed to Astro.
```
Comment (jwitz) [inline astro/cloud-ide/deploy-project.md:121]:
```suggestion
After configuring your repository, commits from any source to your `main` or `dev` branches are automatically deployed to Astro.
```
The workflow doesn't differentiate between Cloud IDE commits and local commits right?

File: astro/cloud-ide/deploy-project.md [hunk 4]
```diff
@@ -34,101 +34,107 @@ Astro Cloud IDE projects use the same structure as Astro projects created with t
...
+| ------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [GitHub](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) | [PAT scopes](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes): `repo` and `workflow`. `workflow` is optional for GitHub repositories that already have a [Cloud IDE GitHub action](#deploy-a-project-from-a-git-repository-to-astro).                   |
+| [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html)                                             | <ul><li> [User role](https://docs.gitlab.com/ee/user/permissions.html#project-members-permissions): `Developer` </li><li> [PAT scope](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#personal-access-token-scopes): `api` </li></ul> |
```
Comment (jlaneve) [inline astro/cloud-ide/deploy-project.md:52]: nit: one of these has bullet points and one doesn't, can we standardize?

File: astro/cloud-ide/deploy-project.md [hunk 5]
```diff
@@ -34,101 +34,107 @@ Astro Cloud IDE projects use the same structure as Astro projects created with t
...
+| Git vendor                                                                                                                | PAT permissions                                                                                                                                                                                                                                           |
+| ------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [GitHub](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) | [PAT scopes](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes): `repo` and `workflow`. `workflow` is optional for GitHub repositories that already have a [Cloud IDE GitHub action](#deploy-a-project-from-a-git-repository-to-astro).                   |
```
Comment (jwitz) [inline astro/cloud-ide/deploy-project.md:51]:
```suggestion
| [GitHub](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) | <ul><li> [PAT scopes](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes): `repo` and `workflow`. `workflow` is optional for GitHub repositories that already have a [Cloud IDE GitHub action](#deploy-a-project-from-a-git-repository-to-astro). </li></ul>                   |
```

File: astro/cloud-ide/deploy-project.md [hunk 6]
```diff
@@ -34,101 +34,107 @@ Astro Cloud IDE projects use the same structure as Astro projects created with t
...
-3. In the **Commit Message** field, enter a commit message. This is the commit message for changes committed  from the Astro Cloud IDE to GitHub.
+3. In the **Commit Message** field, enter a commit message. This is the commit message for changes committed from the Astro Cloud IDE to your Git vendor.
 4. Click **Commit**.
 
-## Deploy a project from GitHub to Astro
+When you first make a commit to your repository, the Cloud IDE commits an additional workflow file that includes steps for pushing your code to Astro. You can then configure your GitHub repository to push your code to a Deployment whenever you commit to specific branches from the Astro Cloud IDE.
+
+You can commit changes from the Astro Cloud IDE to your GitHub repository without configuring destination Deployments in GitHub. However, if you don't configure any destination Deployments, the workflow fails and your changes aren't deployed to Astro. See [Deploy a project from a Git repository to Astro](#deploy-a-project-from-a-git-repository-to-astro) to learn how to deploy to Astro from the Astro Cloud IDE.
```
Comment (jwitz) [inline astro/cloud-ide/deploy-project.md:81]:
```suggestion
You can commit changes from the Astro Cloud IDE to your Git repository without configuring destination Deployments. However, if you don't configure any destination Deployments, the workflow fails and your changes aren't deployed to Astro. See [Deploy a project from a Git repository to Astro](#deploy-a-project-from-a-git-repository-to-astro) to learn how to deploy to Astro from the Astro Cloud IDE.
```

**Pull request merged**